### ISSUE TITLE
Calling a SharePoint REST API Endpoint

### ISSUE THREAD

**Issue by qazqwertish** [OfficeDev/outlook-dev-docs#488]
In the action processing phase, you specify that our service should verify the bearer token, but, in this case we are using a SharePoint REST API endpoint. The behavior is:
- When making the call, we receive a 401 Unauthorized.
- If we include the header like it is specified in this post ("headers": [ { "name": "Authorization", "value": "" } ]) we receive a 403 Forbidden.

Is there a direct way to make a call to the SharePoint REST API from an Actionable Message? We need to call that endpoint using the current user that received the email.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: da46b6c4-8c26-e5a8-fafc-f5fa9549740c
* Version Independent ID: 2ccd32e5-2e1c-470f-8e9f-7e8aa2f64a00
* Content: [Security requirements for actionable messages - Outlook Developer](https://docs.microsoft.com/en-us/outlook/actionable-messages/security-requirements)
* Content Source: [docs/actionable-messages/security-requirements.md](https://github.com/OfficeDev/outlook-dev-docs/blob/master/docs/actionable-messages/security-requirements.md)
* Product: **outlook**
* GitHub Login: @jasonjoh
* Microsoft Alias: **jasonjoh**

**Comment by jasonjoh** [comment]
No, you can't call it directly. The problem is that you need an access token compatible with the SharePoint REST API with whatever appropriate permission scopes you need for that particular API, and the token that we send isn't setup for that.

What you should do is have a backend service/web API that your actionable message targets, then have your service make the SharePoint REST call.

**Comment by qazqwertish** [comment]
Thank you very much for your help @jasonjoh. 

We thought about that solution, but the endpoints we need (i.e. the Like action in a post) depend on the user itself and we cannot "like" a post on behalf of another user. The endpoint expects only the ID of the post and whoever makes the call will be the author of that action.

The other solution we thought was to have just a link in the email that will open the site itself and execute the action, where the user will be already logged in, but this solution makes a bit pointless to use Actionable Messages.

**Comment by johnnycardy** [comment]
If your service already has an access token or refresh token for your SharePoint app stored then you could use that (looking them up via the supplied user email address, which can be trusted). But if the user hasn't already authenticated with you/granted permissions etc, then the link won't work and the user would have to go through a login process.
